### PR TITLE
fix: Conditional Rules -> "Add Other"

### PR DESCRIPTION
**Pull request by ShadeWyrm**
Completes #3975

Note: This does not finish the other related issue #3974....

To test easily:
-> Create a Radio Field
-> Click "Edit Rules"
-> Click the new "Add 'Other'" button!~

It adds the new Other field to the rules view, and the template.

### CODE REVIEW

**Comment by timarney** [comment]
Icon should fill white for hover and focus states

Screenshot 2024-07-15 at 2 13 39 PM

Screenshot 2024-07-15 at 2 11 24 PM

Remove the fill hardcoded on the svg itself.

on parent button add

```
group/button
```

i.e. on icon add 
`group-hover/button:fill-white group-focus/button:fill-white`

File: app/(gcforms)/[locale]/(form administration)/form-builder/components/shared/conditionals/AddOther.tsx [hunk 1]
```diff
@@ -46,16 +58,25 @@ export const AddOther = ({ item }: { item: FormElementWithIndex }) => {
...
-      add(item.index, FormElementTypes.textField, data, groupId);
+      itemId = await add(item.index, FormElementTypes.textField, data, groupId);
     } else {
-      add(item.index, FormElementTypes.textField, data);
+      itemId = await add(item.index, FormElementTypes.textField, data);
     }
+
+    const newRule = { elementId: `${itemId}`, choiceId: `${item.id}.${lastChoice}` };
+    OnComplete(newRule);
   }, [add, item, groupId]);
```
Comment (timarney) [inline app/(gcforms)/[locale]/(form administration)/form-builder/components/shared/conditionals/AddOther.tsx:70]: See lint error here

Comment (ShadeWyrm) [reply app/(gcforms)/[locale]/(form administration)/form-builder/components/shared/conditionals/AddOther.tsx:70]: I'm not getting any lint error flagged? What are you seeing?

Comment (timarney) [reply app/(gcforms)/[locale]/(form administration)/form-builder/components/shared/conditionals/AddOther.tsx:70]: https://github.com/cds-snc/platform-forms-client/pull/4002/files#diff-0087593d121005500352fe89851cd5469758f2016edaef5a81a0beda6343995eR69

File: app/(gcforms)/[locale]/(form administration)/form-builder/components/shared/conditionals/AddOther.tsx [hunk 2]
```diff
@@ -71,7 +71,11 @@ export const AddOther = ({
 
   return (
     <>
-      <Button className="!m-0 !mt-4 inline" theme={"secondary"} onClick={addOther}>
+      <Button
```
Comment (timarney) [inline app/(gcforms)/[locale]/(form administration)/form-builder/components/shared/conditionals/AddOther.tsx]: Pushed a commit here for you

https://github.com/cds-snc/platform-forms-client/pull/4002/commits/fee466d0886a086bb1a30cf5908afae8fea67f9c

**Review by timarney — DISMISSED** [review]

**Pull request merged**